### ISSUE TITLE
Restrict issue to set as completed untill any pending issues exists in "Blocked By"

### ISSUE THREAD

**Issue by SumonT2** [makeplane/plane#2572]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Currently, I can create an issue and link it to another issue that's blocking current. However, I'd like to propose an enhancement where, in order to mark the current issue as completed, the linked issue that's blocking it must also be marked as completed. This ensures that dependencies are resolved before closing the current issue.

### Why should this be worked on?

As described on Summary.

**Comment by vihar** [comment]
Hello @SumonT2, the idea is great. Do you have any suggestions for improving the user experience? For instance, should we display alerts, redirect users to the blocking issue, or completely disable the state until the blocked issue is resolved? Your insights will greatly help. Thanks.

**Comment by prathamjagga** [comment]
Can you share the screenshots or add the steps to reproduce, I cant understand properly.

**Comment by vihar** [comment]
Workflows let you restrict which state transitions are allowed and show a blocker message when someone attempts a move you haven't permitted: https://docs.plane.so/workflows-and-approvals/workflows — Business, with transition conditions on Enterprise Grid.

What isn't there is tying that to blocked-by relations, which is the specific thing asked for here. That isn't on our roadmap.

Closing this out.
